compiler/noirc_frontend/src/hir/comptime/interpreter/builtin/builtin_helpers.rs

```diff
@@ -93,6 +98,47 @@ pub(crate) fn get_array(
     }
 }
 
+/// Get the fields if the value is a `Value::Struct`, otherwise report that a struct type
+/// with `name` is expected. Returns the `Type` but doesn't verify that it's called `name`.
+pub(crate) fn get_struct_fields(
+    name: &str,
+    (value, location): (Value, Location),
+) -> IResult<(HashMap<Rc<String>, Value>, Type)> {
+    match value {
+        Value::Struct(fields, typ) => Ok((fields, typ)),
+        _ => {
+            let expected = StructType::new(
+                StructId::dummy_id(),
+                Ident::new(name.to_string(), location.span),
+                location,
+                Vec::new(),
+                Vec::new(),
+            );
+            let expected = Type::Struct(Shared::new(expected), Vec::new());
+            type_mismatch(value, expected, location)
```

Comment:
The return type is actually different in this case, it takes `EllipticCurvePoint` for example and returns `[Field; 3]`. 
I suppose we can look up the function attributes, although this would be the only function what needs it so far. 

I did it this way because for example `get_array` and all the other `get_<thing>` functions in `builtin_helpers` that take the `interner` seem to generate some kind of generic type like this:
```
let type_var = Box::new(interner.next_type_variable());
let expected = Type::Array(type_var.clone(), type_var);
type_mismatch(value, expected, location)
```
I thought you just want the general idea and the location.

Reply:
Approximating the types in the other cases is usually fine, it's just a bit worse in the case of structs since you'd be hard coding the struct name, generic counts, and can panic if the dummy id leaks. Since we don't pass in the argument types at all though it'll have to do.

Reply:
> It should be retrievable from either the function's parameters

Just so I understand: we could pass in the parameters from the `FuncMeta` that we can still look up in `call_special`, however if the argument is actually _not_ a `Value::Struct`, then wouldn't it be reasonable to assume that the type in the function parameter will also not be a `Type::Struct`? Otherwise the compiler would have rejected it already. By hardcoding the struct name (and the fields), we're a bit more suggestive of the assumptions on the interpreter side, otherwise it would say it expected exactly what it got, but it's not a struct.